packages/core/test/lib/request.test.ts

```diff
@@ -0,0 +1,40 @@
+import { Event } from '@sentry/types';
+
+import { API } from '../../src/api';
+import { eventToSentryRequest } from '../../src/request';
+
+describe('eventToSentryRequest', () => {
+  const api = new API('https://dogsarebadatkeepingsecrets@squirrelchasers.ingest.sentry.io/12312012');
+  const event: Event = {
+    contexts: { trace: { trace_id: '1231201211212012', span_id: '12261980', op: 'pageload' } },
+    environment: 'dogpark',
+    event_id: '0908201304152013',
+    release: 'off.leash.park',
+    spans: [],
+    transaction: '/dogs/are/great/',
+    type: 'transaction',
+    user: { id: '1121', username: 'CharlieDog', ip_address: '11.21.20.12' },
+  };
+
+  it('adds sampling information to transaction item header', () => {
+    event.tags = { samplingMethod: 'client_rate', sampleRate: '0.1121', dog: 'Charlie' };
+
+    const result = eventToSentryRequest(event as Event, api);
+
+    const [envelopeHeaderString, itemHeaderString, eventString] = result.body.split('\n');
+
+    const envelope = {
+      envelopeHeader: JSON.parse(envelopeHeaderString),
+      itemHeader: JSON.parse(itemHeaderString),
+      event: JSON.parse(eventString),
+    };
+
+    // the right stuff is added to the item header
+    expect(envelope.itemHeader).toEqual({ type: 'transaction', sample_rates: [{ id: 'client_rate', rate: '0.1121' }] });
```

Comment:
If we do not use tags to pass data around, is there a fundamental reason to report the sampling rate as a string?

Reply:
No - just didn't want to change the datatype for tags. (Though, as an aside, I don't entirely see why we've said that a tag _can't_ have a number as its value - relay accepts a number just fine.)

Reply:
> relay accepts a number just fine

And converts to string.

It uses a ["lenient string" and converts a bunch of other input types internally](https://github.com/getsentry/relay/blob/31a7c1ecedf3f061562f0a489472ec0c3b337dbb/relay-general/src/protocol/types.rs#L720-L802).

Reply:
Right, that's my point. It's perfectly fine for us to send a number to relay, so why does our type disallow it?

Reply:
Relay is lenient; the SDKs don't have to be.

We're talking about tags, but the line here is an envelope item header... I think we went off-topic :)

Reply:
But there's also no reason for the SDK's _not_ to be in this case.

And yes, it's a tangent. But you took the bait. 😉